sky/catalog/data_fetchers/fetch_runpod.py

```diff
@@ -0,0 +1,466 @@
+        oneWeekPrice
+        communitySpotPrice
+        secureSpotPrice
+        lowestPrice(input: {{gpuCount: {gpu_count}}}) {{
+          minimumBidPrice
+          uninterruptablePrice
+          minVcpu
+          minMemory
+          stockStatus
+          compliance
+          maxUnreservedGpuCount
+          availableGpuCounts
+        }}
+      }}
+    }}
+    """
+
+    result = graphql.run_graphql_query(query)
+
+    if 'errors' in result:
+        raise RuntimeError(f'GraphQL errors: {result["errors"]}')
+
+    try:
+        gpu_type = result['data']['gpuTypes'][0]
+    except Exception as e:
+        error_msg = ('No GPU Types found in RunPod query with'
+                     f'gpu_id={gpu_id}, gpu_count={gpu_count}')
+        raise ValueError(error_msg) from e
+
+    return gpu_type
+
+
+def format_price(price: float) -> float:
+    """Format price to two decimal places."""
+    return round(price, 2)
+
+
+def format_gpu_name(gpu_type: Dict[str, Any]) -> str:
+    """Format GPU name to match the required format.
+
+    Programmatically generates the name from RunPod's GPU display name.
+    For compatibility, some names are overridden in GPU_NAME_OVERRIDES.
+    """
+    # Extract base name
+    base_name = (
+        gpu_type['displayName']
+        # handle formatting names of RTX GPUs
+        .replace('RTX PRO ', 'RTXPRO')
+        # skypilot has no hyphen in RTX names. ie. RTX3090, not RTX-3090
+        .replace('RTX ', 'RTX')
+        # replace spaces with hyphens
+        .replace(' ', '-'))
+
+    # handle name overrides for backwards compatibility
+    if base_name in GPU_NAME_OVERRIDES:
+        base_name = GPU_NAME_OVERRIDES[base_name]
+
+    return base_name
+
+
+def get_gpu_info(gpu_type: Dict[str, Any],
+                 gpu_count: int) -> Optional[Dict[str, Any]]:
+    """Extract relevant GPU information from RunPod GPU type data."""
+    gpu_name = format_gpu_name(gpu_type)
+    vcpus = DEFAULT_GPU_INFO.get(gpu_name, {}).get('vcpus')
+
+    # Use minVcpu lowestPrice if defaults not available
+    # Don't use this value by default as it is dynamic and changes often
+    if vcpus is None:
+        # Returns vcpus already scaled by the GPU count
+        vcpus = gpu_type.get('lowestPrice', {}).get('minVcpu')
+    else:
+        vcpus = vcpus * gpu_count
+
+    # This is the VRAM memory per GPU, not the CPU memory
+    memory = gpu_type.get('memoryInGb')
+
+    # Return None if memory or vcpus not valid
+    if not isinstance(vcpus, (float, int)) or vcpus <= 0:
+        print(f'Skipping GPU {gpu_name}:'
+              ' vCPUs must be a positive number, not {vcpus}')
+        return None
+    if not isinstance(memory, (float, int)) or memory <= 0:
+        print(f'Skipping GPU {gpu_name}:'
+              ' Memory must be a positive number, not {memory}')
+        return None
+
+    # Convert the counts, vCPUs, and memory to float
+    # for consistency with skypilot's catalog format
+    return {
```

Comment:
NOTE: you can use typed dicts from the typing library to enforce standard keys and value types, which is likely going to be helpful in a case liek this where you're passing around dicts instead of objects. I'd advise looking more into it